core/interfaces.go

```diff
@@ -262,6 +262,9 @@ type Storage interface {
 	UpdateApplicationOrganization(context storage.TransactionContext, applicationOrganization model.ApplicationOrganization) error
 
 	InsertAPIKey(context storage.TransactionContext, apiKey model.APIKey) (*model.APIKey, error)
+
+	CheckAuthTypesExists() (bool, error)
```

Comment:
Let's make these 2 functions to be generic, i.e. it should have a parameter collectionName.